src/clusterfuzz/_internal/bot/tasks/utasks/uworker_handle_errors.py

```diff
@@ -12,100 +12,79 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Module for handling errors in utasks."""
-from clusterfuzz._internal.bot.tasks import setup
-from clusterfuzz._internal.bot.tasks.utasks import analyze_task
-from clusterfuzz._internal.bot.tasks.utasks import fuzz_task
-from clusterfuzz._internal.bot.tasks.utasks import minimize_task
-from clusterfuzz._internal.bot.tasks.utasks import progression_task
-from clusterfuzz._internal.bot.tasks.utasks import regression_task
-from clusterfuzz._internal.bot.tasks.utasks import symbolize_task
-from clusterfuzz._internal.bot.tasks.utasks import variant_task
+
+from typing import Callable
+from typing import Dict
+
 from clusterfuzz._internal.protos import uworker_msg_pb2
 
+# A handler takes an output object with its `error_type` set, and does whatever
+# it wants with it.
+ErrorHandler = Callable[[uworker_msg_pb2.Output], None]
+
+# Must use string type because of protobuf enum shenanigans.
+HandlerDict = Dict['uworker_msg_pb2.ErrorType', ErrorHandler]
+
+
+class CompositeErrorHandler:
+  """A handler for several different types of uworker errors."""
+
+  def __init__(self, handlers: HandlerDict):
+    """Initializes a handler that delegates to the values in `handlers`.
+
+    For example:
+
+      CompositeErrorHandler({FOO: handle_foo, BAR: handle_bar})
+
+    Will delegate handling `FOO` errors to `handle_foo`, and `BAR` errors
+    to `handle_bar`.
+    """
+    self._handlers = handlers
+
+  # Must use string types because `CompositeErrorHandler` is not defined yet.
```

Comment:
I didn't know how to handle this before. Cool!

Reply:
Yup!